.github/setup_evap/action.yml

```diff
@@ -0,0 +1,39 @@
+name: "Setup EvaP Testing Environment"
+description: "Sets up the nix environment with caching and copy localsettings"
+
+inputs:
+  shell:
+    description: "name of development shell to use"
+    required: false
+    default: .#evap-dev
+  start-db:
+    description: "whether or not to run the database in the background"
+    required: false
+    default: false
+  npm-ci:
+    description: "whether or not to run `npm ci`"
+    required: false
+    default: false
+
+runs:
+  using: "composite"
+  steps:
+    - uses: DeterminateSystems/nix-installer-action@main
+    - uses: DeterminateSystems/magic-nix-cache-action@main
```

Comment:
Ah, I saw that [your run took forever](https://github.com/e-valuation/EvaP/actions/runs/11113996889/job/30980148541?pr=2274), I suppose the old cache expired? Or, from the log
> disabling binary cache 'http://127.0.0.1:37515/' for 60 seconds
> error: unable to download 'http://127.0.0.1:37515/v0ywj1c0sr29wgfs2di9j83c33vf3jvz.narinfo': HTTP error 418
>
>   response body:
>
>   GitHub API error: GitHub Actions Cache throttled Magic Nix Cache. Not trying to use it again on this run.

In [this later run](https://github.com/e-valuation/EvaP/actions/runs/11113996889/job/30997973321?pr=2274), setup_evap finished in under 1 minute.

The download of the wheels from pypi seems to be flaky though, I'll check what is going on. In theory, the tests from our dependencies shouldn't be run anyways :thinking:

Reply:
Hmm, yeah, maybe it will sort itself out. I explicitly re-ran the actions to give it a chance to have everything cached. Might have temporarily glitched. The 20 minutes I observed it taking for the stuff to rebuild isn't optimal, but if caching works most of the time, we might be fine.

Resolving for now